spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -822,28 +866,14 @@ function timeoutChannelUpgrade(
     // get underlying connection for proof verification
     connection = getConnection(currentChannel.connectionIdentifier)
 
-    // counterparty channel must be proved to still be in OPEN state or INITUPGRADE state (crossing hellos)
-    abortTransactionUnless(counterpartyChannel.State === OPEN || counterpartyChannel.State == INITUPGRADE)
+    // counterparty channel must be proved to not have completed flushing after timeout has passed
+    abortTransactionUnless(counterpartyChannel.state !== OPEN || counterpartyChannel.state == FLUSHCOMPLETE)
+    abortTransactionUnless(counterpartyChannel.sequence === currentChannel.sequence)
```

Comment:
Just noting, in ibc-go we would need a proof guarentee. That is, don't allow an INIT if a channel been restored/upgraded in the same block